src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -770,21 +770,21 @@ protected Task ProduceEnd()
         {
             ProduceStart(appCompleted: true);
 
-            // Force flush
-            await SocketOutput.FlushAsync();
```

Comment:
You can remove `ProduceEndAwaited` now. Change 

             if (!HasResponseStarted)
             {
                 return ProduceEndAwaited();
             }

to

             if (!HasResponseStarted)
             {
                 ProduceStart(appCompleted: true);
             }

in ProduceEnd().